### PR TITLE
Scale/units conversion

### PR DESCRIPTION
**Pull request by ericpre**
This pull request aims at avoiding displaying a 1E-9 m or 1000 nm scale bar on images by converting the scale and units when a file is open or data are cropped. Diffraction pattern, EDS and EELS spectra are also converted.
The convert_units() method of the axes manager allows converting the units manually.

### CODE REVIEW

**Comment by francisco-dlp** [comment]
This is really great! Could you update it with master?

I am not sure if I agree with the automatic unit conversion (e.g. when cropping) as users may like to keep the original units for whatever reason? Same for loading/saving except for the case where a file format requires specific units.

This feature is very EM-centric. Is there any way to make it more general, maybe making use of pint's metric prefixes?

It was in the plans to add `metadata.Signal.units` and use pint to deal with units during operations. I was hoping to get this feature by the release of HyperSpy 1.0, but, as the release should happen today, that's not going to happen.

**Comment by francisco-dlp** [comment]
In particular I was thinking on #546

**Comment by ericpre** [comment]
There are a few things to discuss in this PR.

> automatic unit conversion (e.g. when cropping)

I will add an argument to the crop method with a default that keeps the original unit. Would it be worth adding a preference setting to set the default behavior?

> EM-centric?

Actually, this is exactly that I tried to avoid and I don't see how it is EM-centric. As it is now, a few io reader (tiff and fei) assume a default unit ('nm'), while with this PR, the unit is determined as a function of pixel number and pixel size. For example, it should convert to 'km' when appropriate!!

I will have a look at #546. Would it be simpler to review in a separate PR?

File: anaconda_hyperspy_environment.yml [hunk 1]
```diff
@@ -23,7 +23,9 @@ dependencies:
 - ipywidgets
 - traits
 - traitsui
+- ncurses
```
Comment (francisco-dlp) [inline anaconda_hyperspy_environment.yml]: Why is ncurses back?

File: hyperspy/io_plugins/blockfile.py [hunk 2]
```diff
@@ -126,7 +126,7 @@ def get_header_from_signal(signal, endianess='<'):
     if signal.axes_manager.navigation_dimension == 2:
         NX, NY = signal.axes_manager.navigation_shape
         SX = signal.axes_manager.navigation_axes[0].scale
-        SY = signal.axes_manager.navigation_axes[0].scale
+        SY = signal.axes_manager.navigation_axes[1].scale
```
Comment (francisco-dlp) [inline hyperspy/io_plugins/blockfile.py]: You're actually fixing a bug here, could you submit the fix in a different PR?

**Comment by francisco-dlp** [comment]
When I mentioned that this is too EM-centric I meant that this only performs the job for length units and energy in eV. I've just checked pint's documentation and it seems that their `to_compact` method does part of what this PR implements in a more general way. e.g.

```python
 >>> (20e-9 * ureg.meter).to_compact()
<Quantity(19.999999999999996, 'nanometer')>
```

I feel very uncomfortable with the idea of letting HyperSpy change automatically the units of the axes, but I totally see your point: it's a lot more human friendly to change it for e.g. plotting. Then, why not changing it automatically only where needed e.g.

* For plotting: the plotting functions could plot using the compact units without actually changing the units in the axes. One could even think on doing this automatically when zooming. This behaviour could be controlled with a new entry in preferences.
* Signal slicing could support units e.g. `s.inav[":30. nm"]`. This could be really cool!

Beside,  `DataAxis` could have a method (similar to the one that you've implemented already) to change the units with an option to changing them to the output of pint's compact.

**Comment by francisco-dlp** [comment]
The syntax for slicing could also be improved by using pint itself e.g. `s.inav[:30 * ureg.nanometer]`.

Also, following #1170, could you send this to RELEASE_next_minor?

**Comment by ericpre** [comment]
OK thanks, I understand your point now and it sounds good! :)
to_compact() is exactly, what I was looking for... I don't how I miss it...

**Comment by francisco-dlp** [comment]
This branch was automatically closed when we renamed `master` `RELEASE_next_major` and it cannot be reopened. Could you resubmitted it to RELEASE_next_minor instead?
